File: Java/src/br/com/tavernadoshackers/sistema/Conta.java
package br.com.tavernadoshackers.sistema;

public class Conta {

    // Atributos
    private String nome;
    private double cpf;
    private double saldo;
    private static int idConta;
    private double numeroConta;
    private double agenciaConta;

    // Construtor
    public Conta(String nome, double cpf, double saldo, double numeroConta) {
        this.nome = nome;
        System.out.println(nome + ", sua conta foi criada com sucesso!");
    }


    // Métodos
    public void extrato() {
        System.out.println(this.nome + ", seu saldo é de R$" + this.saldo);
    }

    public void deposita(double valor) {
        this.saldo = valor;
        System.out.println("Deposito realizado com sucesso! Valor: R$" + valor);
    }

    public void saca(double valor) {
        this.saldo -= valor;
        System.out.println("Saque realizado com sucesso! Valor: R$" + valor);
    }

    public void transfere(double valor, Conta destino) {
        if (this.saldo >= valor) {
            this.saldo -= valor;
            destino.saldo += valor;
            System.out.println("Transferência realizada com sucesso! Valor: R$" + valor);
        } else {
            System.out.println("Transferência não pode ser realizada. (Saldo insuficiênte!)");
        }
    }


    // Getters e Setters
    public String getNome() {
        return nome;
    }

    public void setNome(String nome) {
        this.nome = nome;
    }

    public double getCpf() {
        return cpf;
    }

    public void setCpf(double cpf) {
        this.cpf = cpf;
    }

    public double getSaldo() {
        return saldo;
    }

    public void setSaldo(double saldo) {
        this.saldo = saldo;
    }

    public static int getIdConta() {
        return idConta;
    }

    public static void setIdConta(int idConta) {
        Conta.idConta = idConta;
    }

    public double getNumeroConta() {
        return numeroConta;
    }

    public void setNumeroConta(double numeroConta) {
        this.numeroConta = numeroConta;
    }

    public double getAgenciaConta() {
        return agenciaConta;
    }

    public void setAgenciaConta(double agenciaConta) {
        this.agenciaConta = agenciaConta;
    }
}

File: C#/SistemaInterno/Conta.cs
using System;

namespace ByteBank.SistemaInterno
{
    public abstract class Conta
    {
        // Atributos (Getters e Setters)
        public string Nome { get; set; }
        public double Cpf { get; set; }
        public double Saldo { get; set; }


        // Método Construtor
        protected Conta(string nome, double cpf, double saldo)
        {
            Nome = nome;
            Cpf = cpf;
            Saldo = saldo;
        }


        // Métodos
        public virtual void Extrato()
        {
            Console.WriteLine("Nome: " + Nome);
            Console.WriteLine("CPF: " + Cpf);
            Console.WriteLine("Saldo: " + Saldo);
        }

        public void Depositar(double valor)
        {
            Saldo += valor;
            Console.WriteLine(Nome + " o deposito foi efetuado com sucesso! Valor: R$" + valor);
        }

        public void Sacar(double valor)
        {
            if (valor <= Saldo)
            {
                Saldo -= valor;
                Console.WriteLine(Nome + " o saque foi efetuado com sucesso! Valor: R$" + valor);
            }
            else
            {
                Console.WriteLine("O saque não pode ser efetuado.");
            }
        }

        public void Transferir(double valor, Conta destino)
        {
            if (valor <= Saldo)
            {
                Saldo -= valor;
                destino.Saldo += valor;
                Console.WriteLine(Nome + " a transferência foi efetuada com sucesso! Valor: R$" + valor + " para " + destino.Nome);
            }
            else
            {
                Console.WriteLine("A transferência não pode ser efetuada.");
            }
        }

    }
}
